### PR TITLE
tracing: fixes for double-finished traces and missing sampling

### PR DESCRIPTION
**Pull request by dt**
[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4961)

### CODE REVIEW

File: util/tracing/tracer.go [hunk 1]
```diff
@@ -59,24 +59,29 @@ func JoinOrNew(tr opentracing.Tracer, carrier *Span, opName string) (opentracing
 // callback. If the given DelegatingCarrier is nil, a new Span is created.
 // otherwise, the created Span is a child.
 func JoinOrNewSnowball(opName string, carrier *Span, callback func(sp basictracer.RawSpan)) (opentracing.Span, error) {
-	tr := basictracer.New(CallbackRecorder(callback))
+	tr := basictracer.NewWithOptions(defaultOptions(callback))
 	sp, err := JoinOrNew(tr, carrier, opName)
 	if err == nil {
-		sp.SetBaggageItem(Snowball, "1")
 		// We definitely want to sample a Snowball trace.
+		// This must be set *before* SetBaggageItem, as it will otherwise be ignored.
```
Comment (tschottdorf) [inline util/tracing/tracer.go]: s/it/that/

**Comment by tschottdorf** [comment]
LGTM

**Comment by tschottdorf** [comment]
Would you mind filing an issue to pull the trace up into Executor so that it can track txn retries/statement batching?

**Comment by dt** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[util/tracing/tracer.go, line 66 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4961#-KCIV6Kyk6j2-Wa-vx2Y-r1-66:-KCI_UvqNLt9dfK9wGDP:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/0167a70dbb094e37e45d24614fe27c259b646318/util/tracing/tracer.go#L66)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4961)*

**Comment by dt** [comment]
#4962

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4961#-:-KCI_nhgnZFuVJBiDeh4:-531432257)*

**Comment by tschottdorf** [comment]
Thanks @dt.

**Pull request merged**